idaes_examples/notebooks/docs/unit_models/operations/tests/test_eg_h2o_ideal.py

```diff
@@ -0,0 +1,398 @@
+"""
+Author: Brandon Paul
+"""
+import pytest
+from pyomo.environ import (
+    check_optimal_termination,
+    ConcreteModel,
+    Set,
+    value,
+    Var,
+    units as pyunits,
+)
+from pyomo.util.check_units import assert_units_consistent
+from pyomo.common.unittest import assertStructuredAlmostEqual
+
+from idaes.core import Component
+from idaes.core.util.model_statistics import (
+    degrees_of_freedom,
+    fixed_variables_set,
+    activated_constraints_set,
+)
+from idaes.core.solvers import get_solver
+
+from idaes.models.properties.modular_properties.base.generic_property import (
+    GenericParameterBlock,
+)
+
+from idaes.models.properties.modular_properties.state_definitions import FpcTP
+
+from idaes_examples.notebooks.docs.unit_models.operations.eg_h2o_ideal import config_dict
+
+from idaes.models.properties.tests.test_harness import PropertyTestHarness
+
+
+# -----------------------------------------------------------------------------
+# Get default solver for testing
+solver = get_solver()
+
+
+def _as_quantity(x):
+    unit = pyunits.get_units(x)
+    if unit is None:
+        unit = pyunits.dimensionless
+    return value(x) * unit._get_pint_unit()
+
+
+class TestEGProdIdeal(PropertyTestHarness):
+    def configure(self):
+        self.prop_pack = GenericParameterBlock
+        self.param_args = config_dict
+        self.prop_args = {}
+        self.has_density_terms = True
+
+
+class TestParamBlock(object):
+    @pytest.mark.unit
+    def test_build(self):
+        model = ConcreteModel()
+        model.params = GenericParameterBlock(**config_dict)
+
+        assert isinstance(model.params.phase_list, Set)
+        assert len(model.params.phase_list) == 1
+        for i in model.params.phase_list:
+            assert i in ["Liq",]
+        assert model.params.Liq.is_liquid_phase()
+
+        assert isinstance(model.params.component_list, Set)
+        assert len(model.params.component_list) == 2
+        for i in model.params.component_list:
+            assert i in ["water","ethylene_glycol"]
+            assert isinstance(model.params.get_component(i), Component)
+
+        assert isinstance(model.params._phase_component_set, Set)
+        assert len(model.params._phase_component_set) == 2
+        for i in model.params._phase_component_set:
+            assert i in [
+                ("Liq", "water"),
+                ("Liq", "ethylene_glycol"),
+            ]
+
+        assert model.params.config.state_definition == FpcTP
+
+        assertStructuredAlmostEqual(
+            model.params.config.state_bounds,
+            {
+                "flow_mol_phase_comp": (0, 100, 1000, pyunits.mol / pyunits.s),
+                "temperature": (273.15, 298.15, 450, pyunits.K),
+                "pressure": (5e4, 1e5, 1e6, pyunits.Pa),
+            },
+            item_callback=_as_quantity,
+        )
+
+        assert model.params.pressure_ref.value == 1e5
```

Comment:
It is better to do `value(x)` instead of `x.value`. The second only works for `Vars`, so you might run into issues in cases where something is an `Expression`.

Also - what are you actually testing for here? These are input parameters, so you probably don;t need to test these (that gets handled in the core framework tests).